aptos-move/aptos-replay-benchmark/src/overrides.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Defines different overrides for on-chain state used for benchmarking. With overrides, past
+//! transactions can be replayed on top of a modified state, and we can evaluate how it impacts
+//! performance or other things.
+
+use aptos_logger::error;
+use aptos_types::{
+    on_chain_config::{FeatureFlag, Features, OnChainConfig},
+    state_store::{state_key::StateKey, state_value::StateValue, StateView},
+};
+use serde::Serialize;
+use std::collections::HashMap;
+
+/// Stores feature flags to enable/disable, essentially overriding on-chain state.
+pub struct OverrideConfig {
+    enable_features: Vec<FeatureFlag>,
+    disable_features: Vec<FeatureFlag>,
+}
+
+impl OverrideConfig {
+    pub fn new(enable_features: Vec<FeatureFlag>, disable_features: Vec<FeatureFlag>) -> Self {
+        Self {
+            enable_features,
+            disable_features,
+        }
+    }
+
+    pub(crate) fn get_state_override(
+        &self,
+        state_view: &impl StateView,
+    ) -> HashMap<StateKey, StateValue> {
```

Comment:
I think one can argue that we should return a pair of state override key and value, since it's always one. HashMap makes this aspect confusing, and also the whole override potentially could include more things than features, I suppose (now or later)

Reply:
I plan to extend it beyond features, hence the hashmap. Ideally reading from the path where we preciously generated and saved overrides for state keys, e.g. framework code

Reply:
Gas feature version is also interesting, but affects the behaviour more. The plan is to allow users to not consider outputs as different if diffs only contain gas payment related changes.

Reply:
If we aren't going to extend immediately, it makes sense to have restricted API then extend with that PR, but feel free to keep as well, I don't think it's too important here. My main concern was types for partial and complete overrides being the same can be confusing and better to avoid as long as possible

Reply:
What do you mean by partial / complete override?

Reply:
imagine we have multiple pieces that can override the state. Say I had two functions, for different use-cases / reasons, and one of them only produced a single key-value pair. It seems to me safer to return (key, value) from that API as there is no chance to confuse that (partial) override with the final collection that's used to override (HashMap). But if there are also APIs that return similar collections, etc, the distinction becomes not that clear and useful.

Reply:
Ah I see! Let's land this - I am about to create a follow up anyway to read override from state, so we can override framework, so I will rework some of those APIs. Basically, I think we can have an option to 
1. Override features (CLI)
2. Override gas feature version (CLI)
3. Override compiled modules (point to compiled move package) (CLI)